pages/portal/[portalLink].js

```diff
@@ -2,20 +2,88 @@ import { useState, useEffect } from 'react';
 import { useRouter } from 'next/router';
 
 import { Layout } from '../../components/Layout';
+// import { Loader } from '../../components/Loader';
+import { FeaturedArticle } from '../../components/home/FeaturedArticle';
+import { getListOfAllArticlesWithPortal } from '../../firebase/api/portals';
+import { ArticleExcerpt } from '../../components/ArticleExcerpt';
 
 const Portal = () => {
   const router = useRouter();
 
   const [nameRouter, setNameRouter] = useState();
+  const [listOfArticlesWithPortal, setListOfArticlesWithPortal] = useState([]);
 
   useEffect(() => {
     const { portalLink } = router.query;
     setNameRouter(portalLink);
   }, [router]);
 
+  useEffect(() => {
+    const fetchArticlesWithPortal = async () => {
+      setListOfArticlesWithPortal(
+        await getListOfAllArticlesWithPortal(nameRouter),
+      );
+    };
+
+    if (nameRouter) {
+      fetchArticlesWithPortal();
+    }
+    /*
+    if (!nameRouter) {
+      return <Loader />;
+    }
+    */
+  }, [nameRouter]);
```

Comment:
```suggestion
// this useEffect makes no sense. I would remove it
  useEffect(() => {
    const { portalLink } = router.query;
    setNameRouter(portalLink);
  }, [router]);

  useEffect(() => {
    const fetchArticlesWithPortal = async () => {
      setListOfArticlesWithPortal(
        await getListOfAllArticlesWithPortal(router.query.portalLink),
      );
    };

    if (router.query.portalLink) {
      fetchArticlesWithPortal();
    }

  }, [router.query.portalLink]);
```

    if (!router.query.portalLink) {
      return <Loader />;
    }